cdm/src/main/java/ucar/nc2/dataset/EnhanceScaleMissingImpl.java

```diff
@@ -133,10 +133,15 @@
         return;
     }
 
-    // the other possibility is that you want to apply scale and offset to a signed value, then declare the result unsigned
-    // this.isUnsigned = (orgVar != null) ? orgVar.isUnsigned() : forVar.isUnsigned();
-    this.isUnsigned = isUnsigned(forVar);  // LOOK switch to unsigned var
-    this.convertedDataType = forVar.getDataType().withSign(this.isUnsigned);  // only for netcDF !!??
+    // deal with legacy use of attribute with Unsigned = true
+    Attribute unsignedAttrib = forVar.findAttributeIgnoreCase(CDM.UNSIGNED);
+    boolean isUnsignedSet = unsignedAttrib != null && unsignedAttrib.getStringValue().equalsIgnoreCase("true");
+    if (isUnsignedSet) {
+      forVar.setDataType(forVar.getDataType().withSign(true));

```

Comment:
Is this correct; should it it be withSign(flase)?

Reply:
It is correct, but `DataType.withSign(boolean)` is needlessly confusing. Intuitively, `withSign(boolean)` ought to produce a *signed* data type, right? But it actually does the opposite.

I've refactored that method to accept a `Signedness` enum instead. See my new commit.